src/cli/rustup_mode.rs

```diff
@@ -1312,10 +1318,28 @@ fn toolchain_link(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
 }
 
 fn toolchain_remove(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
-    for toolchain in m.values_of("toolchain").unwrap() {
-        let toolchain = cfg.get_toolchain(toolchain, false)?;
-        toolchain.remove()?;
+    if m.is_present("regex") {
+        assert!(
+            m.values_of("toolchain").unwrap().len() == 1,
+            "exactly one regex filter must be supplied"
+        );
```

Comment:
This should be a nice error message. How should I do that?

Reply:
Ideally you should put it as a clap validation function then clap will enforce it for us.

Reply:
But as I mention above, it may be better to just never use this flag, and simply accept that clap ought to require at least one toolchain for `remove` and we treat the name as a pattern if it has a `*` in it?

Reply:
It supports multiple patterns now.